src/instamatic/calibrate/calibrate_stage_rotation.py

```diff
@@ -0,0 +1,395 @@
+            f'delay={self.delay}, '
+            f'speed_options={self.speed_options.to_dict()})'
+        )
+
+    def __eq__(self, o: object) -> bool:
+        return isinstance(o, CalibStageRotation) and self.to_dict() == o.to_dict()
+
+    @staticmethod
+    def curve_fit_model(
+        span_speed: Tuple[float, float],
+        alpha_pace: float,
+        alpha_windup: float,
+        delay: float,
+    ) -> float:
+        """Model equation for estimating total rotation time for scipy."""
+        alpha_span, speed = span_speed
+        return (alpha_pace * alpha_span + alpha_windup) / speed + delay
+
+    def _alpha_poly(self, span: float) -> float:
+        """Value of `alpha_pace * alpha_span + alpha_windup` polynomial."""
+        return self.alpha_pace * span + self.alpha_windup
+
+    def span_speed_to_time(self, span: float, speed: float) -> float:
+        """`time` needed to rotate alpha by `span` with `speed`."""
+        return self._alpha_poly(span) / speed + self.delay
+
+    def span_time_to_speed(self, span: float, time: float) -> float:
+        """`speed` that allows to rotate alpha by `span` with `speed`."""
+        return self._alpha_poly(span) / (time - self.delay)
+
+    def speed_time_to_span(self, speed: float, time: float) -> float:
+        """Maximum `span` covered with `speed` in `time` (including delay)."""
+        return (speed * (time - self.delay) - self.alpha_windup) / self.alpha_pace
+
+    def plan_rotation(self, target_pace: float) -> RotationPlan:
+        """Given target pace in sec / deg, find nearest pace, speed, delay."""
+        target_speed = self.alpha_pace / target_pace  # exact speed setting needed
+        nearest_speed = self.speed_options.nearest(target_speed)  # nearest setting
+        nearest_pace = self.alpha_pace / nearest_speed  # nearest in sec/deg
+        total_delay = self.alpha_windup / nearest_speed + self.delay
+        return RotationPlan(nearest_pace, nearest_speed, total_delay)
+
+    @classmethod
+    def from_dict(cls, dict_: dict) -> CalibStageRotation:
+        return cls(**dict_)
+
+    @classmethod
+    def from_file(cls, path: Optional[str] = None) -> CalibStageRotation:
+        if path is None:
+            path = Path(calibration_drc) / CALIB_STAGE_ROTATION
+        try:
+            with open(Path(path), 'r') as yaml_file:
+                return cls.from_dict(yaml.safe_load(yaml_file))
+        except OSError as e:
+            prog = 'instamatic.calibrate_stage_rotation'
+            raise OSError(f'{e.strerror}: {path}. Please run {prog} first.')
+
+    @classmethod
+    def live(cls, ctrl: 'TEMController', **kwargs) -> CalibStageRotation:
+        return calibrate_stage_rotation_live(ctrl=ctrl, **kwargs)
+
+    def to_dict(self) -> dict:
+        return {
+            k: float(v) if isinstance(v, float) else v.to_dict() for k, v in vars(self).items()
+        }
+
+    def to_file(self, path: Optional[str] = None) -> None:
```

Comment:
Based on the code below (line 345), I think it makes more sense to ask for an outdir here:

```suggestion
    def to_file(self, outdir: Optional[str] = None) -> None:
```

Reply:
Agreed and applied, see joint comment below.